component/redis.jsonnet

```diff
@@ -9,25 +9,48 @@ local role = std.parseJson(kap.yaml_load('argocd/manifests/' + params.git_tag +
 local role_binding = std.parseJson(kap.yaml_load('argocd/manifests/' + params.git_tag + '/redis/argocd-redis-rolebinding.yaml'));
 local serviceaccount = std.parseJson(kap.yaml_load('argocd/manifests/' + params.git_tag + '/redis/argocd-redis-sa.yaml'));
 local service = std.parseJson(kap.yaml_load('argocd/manifests/' + params.git_tag + '/redis/argocd-redis-service.yaml'));
+
+
+local redisSpec(image) =
+  local strContains(s, substr) = std.findSubstr(substr, s) != [];
+
+  local volumeMounts(path) = [ {
+    name: 'data',
+    mountPath: path,
+  } ];
+
+  {
+    image: image,
+    imagePullPolicy: 'IfNotPresent',
+  } +
+  if strContains(image, 'bitnami') then
+    {
+      env+: [
+        { name: 'ALLOW_EMPTY_PASSWORD', value: 'yes' },
+        { name: 'REDIS_AOF_ENABLED', value: 'no' },
+        { name: 'REDIS_EXTRA_FLAGS', value: "--save '' --dir '/data'" },
+      ],
+      args: [],
+      volumeMounts: volumeMounts('/bitnami/redis/data'),
+    }
+  else
+    {
+      volumeMounts: volumeMounts('/data'),
+    };
+
 local objects = [
   deployment {
     spec+: {
       template+: {
         spec+: {
-          [if std.startsWith(inv.parameters.facts.distribution, 'openshift') then
-            'securityContext']:: {},
+          securityContext:: {},
```

Comment:
This change is unnecessary (unless it's required for the Bitnami image). The `securityContext` provided by the upstream manifests should work fine on non-Openshift distributions.

Reply:
It is required for the bitnami image. Different user ids. I will switch to upstream if `not openshift AND official redis`.

Reply:
Ideally, you could ensure that the correct user ids are provided as `securityContext` when the bitnami image is used, instead of removing the security context completely (except on OpenShift).

Reply:
Done in b38bb964ad171759f0eea5bca04acb20a9b97c03.